src/main/scala/com/databricks/labs/mosaic/core/index/H3IndexSystem.scala

```diff
@@ -85,7 +86,10 @@ object H3IndexSystem extends IndexSystem with Serializable {
     override def indexToGeometry(index: Long, geometryAPI: GeometryAPI): MosaicGeometry = {
         val boundary = h3.h3ToGeoBoundary(index).asScala
         val extended = boundary ++ List(boundary.head)
-        geometryAPI.geometry(extended.map(geometryAPI.fromGeoCoord), POLYGON)
+        geometryAPI.geometry(
+          extended.map(p => geometryAPI.fromGeoCoord(Coordinates(p.lat, p.lng))),
```

Comment:
@edurdevic I would add a Coordinates(GeoCoord) constructor so that we could keep simpler map expression.
Very minor, and I can do it as a part of some other PR.
I am very biased towards 1-liners :)

Reply:
We could, but that would introduce an H3 dependency from our core functions.
I prefer to keep all the H3 references within this file. Maybe a helper function within this file to shorten the map expression?